app/views/camps/_donate_form.html.erb

```diff
@@ -19,15 +20,15 @@
                         <span class='heart' id='heart5' data-animated-heart='<%= asset_path('hearts/heart_05.png')%>'></span>
                     </div>
                     <b> <%=t :donate_text %>
-                      <%= number_field_tag 'grants', nil, in: 0..current_user.grants, step: 1, value: 1, class:'donate-value form-control' %>
+                      <%= number_field_tag 'grants', nil, in: 0..grants_left, step: 1, value: 1, class:'donate-value form-control' %>
```

Comment:
Assigning local variables in the view is generally frowned upon; often you'll see a helper method for this purpose (or, if not, simply setting the variable in the controller action like
```
@grants_left = current_user.grants_left_for(@event)
```
or
```
# app/helpers/application_helper.rb or app/helpers/events_helper.rb
def grants_left
  @grants_left ||= current_user.grants_left_for(@event)
end
```